omc3/knob_extractor.py

```diff
@@ -292,33 +284,70 @@ def _extract(ldb, knobs_dict: KnobsDict, knob_categories: Sequence[str], time: d
                 continue
 
             LOGGER.info(f"Knob value for {knob} extracted: {value} (unscaled)")
-            knobs[knob].value = value
+            knobs[knob] = value
+
+    return knobs
+
 
+def _extract_and_gather(ldb, knobs_definitions: pd.DataFrame,
+                        knob_categories: Sequence[str],
+                        time: datetime) -> tfs.TfsDataFrame:
+    """
+    Main function to gather data from  the state-tracker.
+
+    Args:
+        ldb (pytimber.LoggingDB): The pytimber database.
+        knobs_definitions (KnobsDict): A mapping of all knob-names to KnobEntries.
+        knob_categories (Sequence[str]): Knob Categories or Knob-Names to extract.
+        time (datetime): The time, when to extract.
+
+    Returns:
+        pd.Dataframe: Contains all the extracted knobs.
+        When extraction was not possible, the value attribute of the respective entry is NAN
+
+    """
+    extracted_knobs = extract(ldb, knob_categories=knob_categories, time=time)
+    undefined_knobs = [knob for knob in extracted_knobs if knob not in knobs_definitions.index]
+    if undefined_knobs:
+        raise KeyError(
+            "The following knob(s) could not be found "
+            f"in the knob-definitions: '{', '.join(undefined_knobs)}'"
```

Comment:
Wondering if this check should be done before extraction, which is a slow process. The mistaken user could have to wait a while before being told about this (let's say because of a typo)